### PR TITLE
Update web references in libraries

### LINKED ISSUE #347: Investigate the experience of upgrading libraries with web dependencies
Ideally, class libraries can move through the first couple upgrade steps (NuGet cleanup, project file update) without breaking. Once the TFM is changed to net5.0/netstandard2.0, breaking changes will be introduced.

Libraries with web dependencies have some particular challenges here, though (which SDK is used, etc.), so we should test that experience and learn whether there are improvements we could/should make for that scenario.

### PR DESCRIPTION
**Pull request by mjrousos**
This updates Upgrade Assistant to use the latest version of try-convert (which won't remove classic ASP.NET dependencies from libraries that keep a NetFx TFM) and adds additional package update analyzers to replace ASP.NET dependencies with ASP.NET Core ones *after* the TFM has been updated.

Fixes #347

### CODE REVIEW

File: Microsoft.DotNet.UpgradeAssistant.sln [hunk 1]
```diff
@@ -20,7 +21,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		LICENSE.txt = LICENSE.txt
 		.github\workflows\publish.yml = .github\workflows\publish.yml
 		README.md = README.md
-		ToolsInstaller.targets = ToolsInstaller.targets
```
Comment (twsouthwick) [inline Microsoft.DotNet.UpgradeAssistant.sln:23]: I wish these would autoupdate or just display everything in a directory

File: src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/ComponentIdentifier.cs [hunk 2]
```diff
@@ -17,7 +17,14 @@ public ProjectComponents GetComponents(IProject project)
             }
 
             var file = project.GetFile();
-            var components = file.IsSdk ? GetSDKProjectComponents(project, file) : GetOldProjectComponents(project, file);
+
+            // SDK-style projects can target .NET Framework and use GAC-referenced app models,
+            // so old project components are checked regardless of SDK status
+            var components = GetOldProjectComponents(project, file);
```
Comment (twsouthwick) [inline src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/ComponentIdentifier.cs:23]: Maybe rename it then as it feels weird keeping the `Old` in the name

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs [hunk 3]
```diff
@@ -36,21 +36,28 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
                 throw new ArgumentNullException(nameof(state));
             }
 
-            if (!project.Components.HasFlag(ProjectComponents.AspNetCore))
+            // This reference only needs added to ASP.NET Core exes
```
Comment (twsouthwick) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs:39]: good catch

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs [hunk 4]
```diff
@@ -36,21 +36,28 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
...
+            if (!project.Components.HasFlag(ProjectComponents.AspNetCore) || project.OutputType != ProjectOutputType.Exe)
```
Comment (twsouthwick) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs:40]: nit: all the negation here feels hard to weird....

**Review by twsouthwick — APPROVED** [review]
It may be nice to add some tests for your new packageanalyzer

File: Microsoft.DotNet.UpgradeAssistant.sln (hunk 1 above)
Comment (mjrousos) [reply Microsoft.DotNet.UpgradeAssistant.sln:23]: I agree

File: src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/ComponentIdentifier.cs (hunk 2 above)
Comment (mjrousos) [reply src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/ComponentIdentifier.cs:23]: Good point. I'll rename it.

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs (hunk 4 above)
Comment (mjrousos) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs:40]: I'll De Morgan it to have less !.

**Comment by mjrousos** [comment]
Good call on the test. I'll add a unit test.

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs [hunk 5]
```diff
@@ -80,7 +80,7 @@ protected override async Task<UpgradeStepInitializeResult> InitializeImplAsync(I
             var completeChecks = _orderedProjects.Select(async p => IsCompleted(context, p) || !await RunChecksAsync(p, token).ConfigureAwait(false));
```
Comment (sunandabalu) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs:80]: @mjrousos While you are here, could you address the PR build failures we are seeing - https://dev.azure.com/dnceng/public/_build/results?buildId=1041902&view=logs&j=396e4757-3dc0-597a-e469-5feb3123c7be&t=44c646e6-db1b-5521-5b76-c5c2c57c3a93&l=83 - add a suppression for context.entrypoint check in LN 65.

Comment (mjrousos) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs:80]: Yep, that's what I'm doing in this file. Unless I'm mistaken, the build break came from line 83 checking `entryPoint` for null unnecessarilly, which this change addresses.

Comment (sunandabalu) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs:80]: Ah perfect, I missed the removal of ? :)

**Review by sunandabalu — APPROVED** [review]

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs (hunk 5 above)
Comment (twsouthwick) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs:80]: Yay! I'm glad there's a check for that.

Comment (twsouthwick) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs:80]: Weird that it just started happening w/o a change that got blocked in CI, though

Comment (mjrousos) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs:80]: Yea, I thought the same thing. This broke when I added the check for EntryPoint not being null in #351. It's weird it didn't fail then. Who knows.

**Pull request merged**